terraform/ecs-task.tf

```diff
@@ -0,0 +1,88 @@
+resource "aws_iam_role" "ecs_task_execution_role" {
+  name = "ecsTaskExecutionRole"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = "sts:AssumeRole"
+        Effect = "Allow"
+        Principal = {
+          Service = "ecs-tasks.amazonaws.com"
+        }
+      }
+    ]
+  })
+}
+
+resource "aws_iam_role_policy_attachment" "ecs_task_execution_role_policy" {
+  role       = aws_iam_role.ecs_task_execution_role.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy"
+}
```

Comment:
This will cover most of the permissions your task will need to execute but you'll need to attach an additional policy with specific EC2 related permissions as well. You'll see in the AWS console when your task fails to execute which permissions might be missing but it will take a few iteration cycles to capture them all in a policy.